modules/shared/src/main/scala/org/tessellation/ext/collection/SetRefUtils.scala

```diff
@@ -0,0 +1,18 @@
+package org.tessellation.ext.collection
+
+import cats.effect.{Ref, Sync}
+import cats.syntax.functor._
+
+object SetRefUtils {
+  implicit class RefOps[F[_]: Sync, K, V](val ref: Ref[F, Set[K]]) {
```

Comment:
shouldn't it be Async?

Reply:
If the functions in the `RefOps` require `Sync`, why require `Async` from the outside? I guess it will not be utilized anyway.

Reply:
to be consistent within the repository. at the very beginning we agreed that Async would be the base type for our effects (instead of Sync that we used with cats effect 2)